### PR TITLE
Gh 3130 - CustomAuth login

### PR DESCRIPTION
**Pull request by biafra23**
Handles #3130, #3138 

Changes proposed in this pull request:
- When the user clicks on Continue with Google, thew Web3auth google flow starts
- The redirect url is only handled if it matches the prefix
- At the end in the success case, the is a private key.

### CODE REVIEW

File: Multisig/UI/Safe Management/Create Safe/CreateSafeWithSocialIntroViewController.xib [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <device id="retina6_12" orientation="portrait" appearance="light"/>
     <dependencies>
         <deployment identifier="iOS"/>
-        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="21679"/>
+        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="21678"/>
```
Comment (MouazAlzahabi) [inline Multisig/UI/Safe Management/Create Safe/CreateSafeWithSocialIntroViewController.xib]: Can you please discard your changes to this file as they are not needed ?

File: Multisig/UI/Safe Management/Create Safe/SafeCreatingViewController.swift [hunk 2]
```diff
@@ -18,6 +18,8 @@ class SafeCreatingViewController: UIViewController {
     @IBOutlet weak var infoView2: InfoView!
     @IBOutlet weak var infoView1: InfoView!
 
+    var privateKey: String!
```
Comment (MouazAlzahabi) [inline Multisig/UI/Safe Management/Create Safe/SafeCreatingViewController.swift]: This is not needed right ?

File: Multisig/Cross-layer/Configuration/Config.Example.xcconfig [hunk 3]
```diff
@@ -176,3 +176,10 @@ ASSETCATALOG_COMPILER_APPICON_NAME  = AppIcon_$(SERVICE_ENV)
 //
 // Also, Firebase Analytics requires clients to add -ObjC linker option.
 OTHER_LDFLAGS = $(inherited) -fprofile-instr-generate -ObjC
+
+// Set in https://console.cloud.google.com/apis/credentials?project=web3auth-safe&supportedpurview=project
```
Comment (MouazAlzahabi) [inline Multisig/Cross-layer/Configuration/Config.Example.xcconfig:180]: I would say we need to define all of those three as environment variables in Jenkins, at least the first two

File: Multisig/App/SceneDelegate.swift [hunk 4]
```diff
@@ -132,6 +136,8 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
     }
 
     func scene(_ scene: UIScene, continue userActivity: NSUserActivity) {
+       // CustomAuth.handle(url: userActivity.webpageURL!)
```
Comment (MouazAlzahabi) [inline Multisig/App/SceneDelegate.swift]: not needed

File: Multisig/App/SceneDelegate.swift [hunk 5]
```diff
@@ -32,6 +33,9 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
 
     // MARK: - Scene Life Cycle
     func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
+
+        guard let _ = (scene as? UIWindowScene) else { return }
```
Comment (MouazAlzahabi) [inline Multisig/App/SceneDelegate.swift]: Not needed

File: Multisig/App/SceneDelegate.swift [hunk 6]
```diff
@@ -177,6 +185,11 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
             return
         }
 
+        // Handle Web3Auth redirect
+        if GoogleWeb3AuthLoginModel.isValid(url: incomingURL.absoluteString) {
```
Comment (MouazAlzahabi) [inline Multisig/App/SceneDelegate.swift]: I would move this handling inside the GoogleWeb3AuthLoginModel

File: Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import Foundation
+import CustomAuth
+import UIKit
+
+class GoogleWeb3AuthLoginModel {
```
Comment (MouazAlzahabi) [inline Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift:5]: I would generalized the name to handle all Web2Auth types, Apple and Google, we have

**Review by MouazAlzahabi — APPROVED** [review]

File: Multisig.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -1208,7 +1208,6 @@
 		04B4B6E325B9A08D00CFE24B /* CardTableViewCell.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = CardTableViewCell.swift; sourceTree = "<group>"; };
 		04B4B6E425B9A08D00CFE24B /* CardTableViewCell.xib */ = {isa = PBXFileReference; lastKnownFileType = file.xib; path = CardTableViewCell.xib; sourceTree = "<group>"; };
 		04B964402734427D00A21EE9 /* ChainSettingsTableViewController.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ChainSettingsTableViewController.swift; sourceTree = "<group>"; };
-		04C22B842A3B25A900F58A15 /* String+Error.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = "String+Error.swift"; sourceTree = "<group>"; };
```
Comment (MouazAlzahabi) [inline Multisig.xcodeproj/project.pbxproj:1211]: Are you sure this file is not needed?

**Review by MouazAlzahabi — CHANGES_REQUESTED** [review]

File: Multisig.xcodeproj/project.pbxproj (hunk 8 above)
Comment (biafra23) [reply Multisig.xcodeproj/project.pbxproj:1211]: I had to delete the content to make the project compile. Will double check with jenkins.

File: Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift (hunk 7 above)
Comment (biafra23) [reply Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift:5]: They might be very different. I will unify, when Apple is sufficiently similar, ok?

File: Multisig/Cross-layer/Configuration/Config.Example.xcconfig (hunk 3 above)
Comment (biafra23) [reply Multisig/Cross-layer/Configuration/Config.Example.xcconfig:180]: Added here: https://padmeamidala.ngrok.io/credentials/store/system/domain/_/

File: Multisig.xcodeproj/project.pbxproj (hunk 8 above)
Comment (MouazAlzahabi) [reply Multisig.xcodeproj/project.pbxproj:1211]: I think it won't build on jenkins like this,

File: Multisig/Cross-layer/Configuration/Config.Example.xcconfig (hunk 3 above)
Comment (MouazAlzahabi) [reply Multisig/Cross-layer/Configuration/Config.Example.xcconfig:180]: I resolved that one, check my changes

**Review by MouazAlzahabi — APPROVED** [review]

**Pull request merged**